Base.rte/Devices/Tools/Medikit/Medikit.lua

```diff
@@ -0,0 +1,49 @@
+function Create(self)
+	self.baseStrength = 20;
+end
+function Update(self)
+	if self.FiredFrame then
+		parent = MovableMan:GetMOFromID(self.RootID);
+		if parent and IsActor(parent) then
+			parent = ToActor(parent);
+			local target;
+			local extend = 0;
+			if parent:GetController():IsState(Controller.AIM_SHARP) then
+				extend = self.Radius;	-- Reach out further
+			end
+			for actor in MovableMan.Actors do
+				local dist = SceneMan:ShortestDistance(self.MuzzlePos, actor.Pos, SceneMan.SceneWrapsX);
+				if dist.Magnitude < (actor.Radius + extend) and actor.Team == self.Team and actor.ID ~= parent.ID then
+					target = actor;
+					break;
+				else
+					target = parent;
+				end
+			end
+			if target and (target.Health < target.MaxHealth or target.TotalWoundCount > 0) then
+				local strength = self.baseStrength + math.ceil(2000 / (1 + math.abs(target.Mass) / 2));
+				if target.Health < target.MaxHealth then
+					target.Health = math.min(target.Health + strength, target.MaxHealth);
+				end
+				if target.TotalWoundCount > 0 then
+					target:RemoveAnyRandomWounds(math.ceil(strength / 7));
+				end
+				target:FlashWhite(50);
+				AudioMan:PlaySound("Base.rte/Sounds/GUIs/SlicePicked.wav", SceneMan:TargetDistanceScalar(self.Pos), false, true, -1);
+
+				local radist = math.ceil(5 + target.Radius / 2);
+				for i = 1, radist do
+					local part = CreateMOPixel("Heal Glow", "Base.rte");
+					local vec = Vector(radist * 2, 0):RadRotate(6.28 / radist * i);
+					part.Pos = target.Pos + Vector(0, -radist / 3):RadRotate(target.RotAngle) + vec;
+					part.Vel = target.Vel * 0.5 - Vector(vec.X, vec.Y) / 4;
+					MovableMan:AddParticle(part);
+				end
+				local cross = CreateMOSParticle("Particle Heal Effect", "Base.rte");
```

Comment:
Why not create this in the Create function? Same thing probably applies to the medic drone script now I think of it.

Reply:
Don't you require a new particle every time it's used? I understand the one-time case such as the medikit but the medic drone creates multiple.

Reply:
Ah yeah, wouldn't work for the medic drone for that reason. Would work for the medikit though I think, since it's single-target, one-use.